fortitude/src/ast.rs

```diff
@@ -161,6 +164,18 @@ impl FortitudeNode<'_> for Node<'_> {
         None
     }
 
+    fn indentation(&self, source_file: &SourceFile) -> String {
+        let src = source_file.to_source_code();
+        let start_byte = TextSize::try_from(self.start_byte()).unwrap();
+        let start_index = src.line_index(start_byte);
+        let start_line = src.line_start(start_index);
+        let start_offset = start_byte - start_line;
+        let line = src
+            .slice(TextRange::new(start_line, start_line + start_offset))
+            .to_string();
+        line.chars().take_while(|&c| c.is_whitespace()).collect()
```

Comment:
We might have some fun with tabs here! (although tabs are technically not valid characters in fortran source)

Reply:
I figured tabs might be present in some codes, and hence I wrote `indentation` to return a string of the current indentation level rather than a count of the number of spaces. That way it can include a mix of tabs and spaces. I _think_ this is the safest way to handle them, but could be totally wrong on that.

On this topic, we should probably have a 'no tabs allowed' rule.